### PR TITLE
Implements Spiess' conical VDF

### PR DESCRIPTION
**Pull request by pedrocamargo**
@janzill , Can  I ask you to check my math and my Cython implementation of the conical?

I did not do the derivative by hand, but the function is really long, and therefore ripe for mistakes

Closes issue #204

### CODE REVIEW

**Comment by pedrocamargo** [comment]
I was too tired last night to think of doing numerical differentiation to test the conical derivative.  I did it today and the original implementation was wrong!!!!  All fixed now.

**Comment by janzill** [comment]
@pedrocamargo will do later today. BTW, I like https://www.sympy.org/en/index.html for symbolic maths.

**Comment by janzill** [comment]
@pedrocamargo derivative agrees with what I get. I like the numerical test you implemented!

File: aequilibrae/paths/conical.pyx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                       double [:] fftime,
+                       double[:] alpha,
+                       double [:] beta,
+                       int cores):
+  cdef long long i
+  cdef long long l = deltaresult.shape[0]
+
+  for i in prange(l, nogil=True, num_threads=cores):
+    if link_flows[i] > 0:
+        deltaresult[i] = fftime[i] * ((alpha[i] / capacity[i]) - (
```
Comment (janzill) [inline aequilibrae/paths/conical.pyx:65]: Actually, should the first minus on line 65 be a plus?

Comment (janzill) [reply aequilibrae/paths/conical.pyx:65]: you get a minus from the square root, then a plus from the power, then a minus from the last inner derivative of (1-x) I think

**Review by janzill — COMMENTED** [review]
I'm not sure about one of the signs in the derivative, otherwise it looks good. Can you double-check this please?

File: aequilibrae/paths/conical.pyx (hunk 1 above)
Comment (janzill) [reply aequilibrae/paths/conical.pyx:65]: Actually, I should have a look at what I did earlier, the derivative of sqrt does not have a minus, you are right, sorry

**Review by janzill — APPROVED** [review]

File: aequilibrae/paths/traffic_assignment.py [hunk 2]
```diff
@@ -273,13 +274,13 @@ def set_vdf_parameters(self, par: dict) -> None:
...
         """
         if self.classes is None or self.vdf.function.lower() not in all_vdf_functions:
-            raise Exception('Before setting vdf parameters, you need to set traffic classes and choose a VDF function')
-        self.__dict__['vdf_parameters'] = par
+            raise Exception("Before setting vdf parameters, you need to set traffic classes and choose a VDF function")
+        self.__dict__["vdf_parameters"] = par
         pars = []
-        if self.vdf.function in ["BPR"]:
-            for p1 in ['alpha', 'beta']:
+        if self.vdf.function in ["BPR", "CONICAL"]:
```
[the remark this answers is not in the document]
Comment (pedrocamargo) [reply aequilibrae/paths/traffic_assignment.py:280]: There is already something like that in the VDF class

File: aequilibrae/paths/traffic_assignment.py [hunk 3]
```diff
@@ -273,13 +274,13 @@ def set_vdf_parameters(self, par: dict) -> None:
...
+            for p1 in ["alpha", "beta"]:
```
[the remark this answers is not in the document]
Comment (pedrocamargo) [reply aequilibrae/paths/traffic_assignment.py:281]: Wouldn't this be sort of a "default value" that would be somewhat hidden?

File: aequilibrae/paths/traffic_assignment.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (janzill) [reply aequilibrae/paths/traffic_assignment.py:280]: right, so we test there, here we only assign parameters. I'm ok with that :)

File: aequilibrae/paths/traffic_assignment.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (janzill) [reply aequilibrae/paths/traffic_assignment.py:281]: With a big warning hopefully :) But agreed, let's be explicit and keep both parameters

**Review by janzill — APPROVED** [review]

**Pull request merged**